src/couch_replicator_changes_reader.erl

```diff
@@ -89,9 +89,19 @@ process_change(#doc_info{id = <<>>} = DocInfo, {_, Db, _, _, _}) ->
         "source database `~s` (_changes sequence ~p)",
         [couch_replicator_api_wrap:db_uri(Db), DocInfo#doc_info.high_seq]);
 
-process_change(#doc_info{} = DocInfo, {_, _, ChangesQueue, _, _}) ->
-    ok = couch_work_queue:queue(ChangesQueue, DocInfo),
-    put(last_seq, DocInfo#doc_info.high_seq);
+process_change(#doc_info{id = Id} = DocInfo, {Parent, Db, ChangesQueue, _, _}) ->
+    case is_doc_id_too_long(byte_size(Id)) of
+        true ->
+            ShortId = lists:sublist(binary_to_list(Id), 64),
+            SourceDb = couch_replicator_api_wrap:db_uri(Db),
+            couch_log:error("Replicator: document id `~s...` from source db "
```

Comment:
@eiri Since this is a cherry-picked commit of the upstream change. Would it make sense to merge this then have a separate PR for both upstream. Or maybe just have a commit for upstream which does this and then add that commit to this branch as well (just to make sure it doesn't diverge from upstream unnecessarily)